src/clients/api/mutations/useApproveToken.ts

```diff
@@ -0,0 +1,44 @@
+import { useMutation, MutationObserverOptions } from 'react-query';
+import { queryClient, approveToken, IApproveTokenInput, ApproveTokenOutput } from 'clients/api';
+import FunctionKey from 'constants/functionKey';
+import { useToken } from 'hooks/useContract';
+
+const useApproveToken = (
+  assetId: string,
+  // TODO: use custom error type
+  options?: MutationObserverOptions<
+    ApproveTokenOutput,
+    Error,
+    Omit<IApproveTokenInput, 'tokenContract'>
+  >,
+) => {
+  const tokenContract = useToken(assetId);
+  return useMutation(
+    FunctionKey.APPROVE_TOKEN,
+    (params: Omit<IApproveTokenInput, 'tokenContract'>) =>
+      approveToken({
+        tokenContract,
+        ...params,
+      }),
+    {
+      ...options,
+      onSuccess: (
+        data: void,
+        variables: Omit<IApproveTokenInput, 'tokenContract'>,
+        context: unknown,
+      ) => {
```

Comment:
Same thing here, React Query infers the types from `approveToken`:
```suggestion
      onSuccess: (data, variables, context) => {
```